Composer/packages/server/src/controllers/publisher.ts

```diff
@@ -53,16 +54,23 @@ export const PublishController = {
   publish: async (req, res) => {
     const target: string = req.params.target;
     const user = await ExtensionContext.getUserFromRequest(req);
-    const { accessToken = '', metadata, sensitiveSettings } = req.body;
+    const { accessToken = '', metadata, sensitiveSettings, publishTarget } = req.body;
     const projectId: string = req.params.projectId;
+    currentPublishTarget = publishTarget;
     const currentProject = await BotProjectService.getProjectById(projectId, user);
 
     // deal with publishTargets not exist in settings
     const publishTargets = currentProject.settings?.publishTargets || [];
     const allTargets = [defaultPublishConfig, ...publishTargets];
 
     const profiles = allTargets.filter((t) => t.name === target);
-    const profile = profiles.length ? profiles[0] : undefined;
+    let profile: PublishTarget;
```

Comment:
The same lines are repeated on line 68. Wrap it in a function?

Reply:
+1 for @srinaath 
also you can simplify it by:
```typescript
let profile: PublishTarget = profiles[0];
if(!profile) {
  profile = publishTarget;
  currentProject.settings?.publishTargets && currentProject.settings?.publishTargets.push(publishTarget);
}
```

Reply:
Wraped in a function